File: MyClub/MyClub.Services/Database/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyClub.Services.Database
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [MaxLength(50)]
        public string BarCode { get; set; } = string.Empty;
        
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        
        public int ColorId { get; set; }
        
        [ForeignKey("ColorId")]
        public virtual Color Color { get; set; }
        
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? UpdatedAt { get; set; }
        
        
        [Column(TypeName = "decimal(5,2)")]
        public decimal? Rating { get; set; }

        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]    
        public virtual Category Category { get; set; }
        public virtual ICollection<ProductAsset> ProductAssets { get; set; } = new List<ProductAsset>();

        public virtual ICollection<ProductSize> ProductSizes { get; set; } = new List<ProductSize>();
    }
}

File: MyClub/UI/myclub_desktop/lib/models/product.dart
import 'package:myclub_desktop/models/category.dart';
import 'package:myclub_desktop/models/color.dart' as model;
import 'package:myclub_desktop/models/asset.dart';
import 'package:myclub_desktop/models/product_size.dart';

class Product {
  final int? id;
  final String? name;
  final String? description;
  final String? barCode;
  final double? price;
  final model.Color? color;
  final bool? isActive;
  final DateTime? createdAt;
  final double? rating;
  final Category? category;
  final Asset? primaryImageUrl;
  final List<Asset>? imageUrls;
  final List<ProductSize>? sizes;

  Product({
    this.id,
    this.name,
    this.description,
    this.barCode,
    this.price,
    this.color,
    this.isActive,
    this.createdAt,
    this.rating,
    this.category,
    this.primaryImageUrl,
    this.imageUrls,
    this.sizes,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      barCode: json['barCode'],
      price: json['price']?.toDouble(),
      color: json['color'] != null ? model.Color.fromJson(json['color']) : null,
      isActive: json['isActive'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      rating: json['rating']?.toDouble(),
      category: json['category'] != null ? Category.fromJson(json['category']) : null,
      primaryImageUrl: json['primaryImageUrl'] != null ? Asset.fromJson(json['primaryImageUrl']) : null,
      imageUrls: json['imageUrls'] != null 
          ? List<Asset>.from(json['imageUrls'].map((x) => Asset.fromJson(x))) 
          : null,
      sizes: json['sizes'] != null 
          ? List<ProductSize>.from(json['sizes'].map((x) => ProductSize.fromJson(x))) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'barCode': barCode,
      'price': price,
      'color': color?.toJson(),
      'isActive': isActive,
      'createdAt': createdAt?.toIso8601String(),
      'rating': rating,
      'category': category?.toJson(),
      'primaryImageUrl': primaryImageUrl?.toJson(),
    };
  }
}
